mariamihaelafainarea/assignment1/Anagrams

```diff
@@ -0,0 +1,188 @@
+            return false;
+        }
+        int length = text1.length();
+       //i store each letter
+       //and i keep in the hashmap also the number of entrances,
+       //the number of occurences of that letter
+        for (int i = 0; i < length; i++) {
+            int numberEntrances = 0;
+            if (table.get(text1.charAt(i)) == null) {  
+                 table.put(text1.charAt(i),1);
+                
+            } else {
+                numberEntrances = table.get(text1.charAt(i)) + 1;
+                table.put(text1.charAt(i),numberEntrances);
+            }     
+            
+        }
+        //then i iterate through the letters of the other word
+        //and i check is the current letter
+        //is also in the hashmap
+        //and i update the number of occurences from hashmap
+        //untill the end of the searching 
+        for (int i = 0; i < length; i++) {
+            int numberEntrances = 0 ;
+            if (table.get(text2.charAt(i)) == null) {
+                return false;
+            }
+            
+            numberEntrances = table.get(text2.charAt(i));
+
+            numberEntrances--;
+            
+            table.replace(text2.charAt(i),numberEntrances);
+
+        }
+        //if the hashmap is empty of occurences
+        //everything is okey
+        for (Character c : table.keySet()) {
+            if(table.get(c) != 0) {
+                return false;
+            }
+        }
+        return true;
+      
+    }
+   
+    private Boolean findIfAnagramSentences(String[] words,String[] words2) {
+        //We have to compare senteces
+        //So we compare first the number of words
+         HashMap<String, Integer> table2 = new HashMap<>();
+        if (words.length != words2.length) {
+            //if the number isn't okey ...the senteces aren't anagrams
+            return false;
+        }
+        for (int i = 0; i < words.length; i++) {
+            //store each word from the first sentence in a hashmap
+            if (table2.get(words[i]) == null ) {
+                table2.put(words[i], 1);
+            } else {
+                //every time it is repeated
+                //i increment the number of entries
+                int numberEntrances = table2.get(words[i]);
+                numberEntrances++;
+                table2.replace(words[i], numberEntrances);
+            } 
+        }
+        for (int i = 0; i < words2.length; i++) {
+            //then iterate through the second sentence
+            //and search for every word
+            Boolean isOkey = false;
+            String anagrama = null;
+            for (String c : table2.keySet()) {
+                //when i'm searching for a word
+                //i keep in mind that it can exist
+                //in the first sentece
+                //but not in this form
+                //but in an anagram
+                if ( findIfAnagramWords(words2[i], c) ) {
```

Comment:
Here you are making an unnecessary number of calls to findIfAnagramWords. O(N^2) where N is the number of words in the sentence. That can get really expensive in some edge cases. Can you find a way to make sure you call this method only O(N) of times?

Reply:
I have sorted the words and added like this in a hashmap and then simply find that words sorted